src/WebJobs.Script.WebHost/ProxyFunctionExecutor.cs

```diff
@@ -22,19 +22,87 @@ public class ProxyFunctionExecutor : IFuncExecutor
     {
         private readonly WebScriptHostManager _scriptHostManager;
         private readonly ISecretManager _secretManager;
+        private readonly string _httpPrefix;
+
         private WebHookReceiverManager _webHookReceiverManager;
 
         internal ProxyFunctionExecutor(WebScriptHostManager scriptHostManager, WebHookReceiverManager webHookReceiverManager, ISecretManager secretManager)
         {
             _scriptHostManager = scriptHostManager;
             _webHookReceiverManager = webHookReceiverManager;
             _secretManager = secretManager;
+
+            _httpPrefix = "api/";
+
+            if (_scriptHostManager.Instance != null)
+            {
+                var json = _scriptHostManager.Instance.ScriptConfig.HostConfig.HostConfigMetadata;
+
+                if (json != null && json["http"] != null && json["http"]["routePrefix"] != null)
+                {
+                    _httpPrefix = json["http"]["routePrefix"].ToString() + "/";
+                }
+            }
         }
 
         public async Task ExecuteFuncAsync(string funcName, Dictionary<string, object> arguments, CancellationToken cancellationToken)
         {
             HttpRequestMessage request = arguments[ScriptConstants.AzureFunctionsHttpRequestKey] as HttpRequestMessage;
-            var function = _scriptHostManager.GetHttpFunctionOrNull(request);
+
+            FunctionDescriptor function = null;
+            var path = request.RequestUri.AbsolutePath.Trim(new char[] { '/' });
+
+            if (path.StartsWith(_httpPrefix))
+            {
+                path = path.Remove(0, _httpPrefix.Length);
+            }
+
+            // This is a call to the local function app, before handing the route to asp.net to pick the FunctionDescriptor the following will be run:
+            // 1. If the request maps to a local http trigger function name then that function will be picked.
+            // 2. Else if the request maps to a custom route of a local http trigger function then that function will be picked
+            // 3. Otherwise the request will be given to asp.net to pick the appropriate route.
+            foreach (var func in _scriptHostManager.HttpFunctions.Values)
+            {
+                if (!func.Metadata.IsProxy)
```

Comment:
I thought this proxy executor is only used for proxies? When would this condition be true?

Reply:
This is our callback logic when the backend is a local function, we go thru the list of FunctionDescriptors, but we don't want this logic for proxy function descriptors. there is some comment above describing the logic.

---
In reply to: [146595677](https://github.com/Azure/azure-webjobs-sdk-script/pull/2060#discussion_r146595677) [](ancestors = 146595677)